owncloud/ios-app
# [feature/fileprovider-passcode-UI] File Provider Passcode UI

## Description
Added a passcode UI for the file provider, when a passcode was set in the app. With this UI the file provider can be used with passcode set immediately. Passcode can be entered directly in the file provider extension.

In addition the resign script and fastlane script got an update to set all identifiers and provisioning profiles.

## Related Issue
#880 

## Motivation and Context
Secure the app with passcode set immediately and and use the file provider extension by unlocking the passcode in the extension.

## How Has This Been Tested?
- Set a passcode
- Lock application `immediately`
- Open Files.app
- Select ownCloud file provider
- Tap `Authorize`
- Passcode UI should appear and disappear after entering the correct passcode
- Check Biometrical unlock
- Check file provider unlock in third party apps

## Screenshots (if appropriate):

https://user-images.githubusercontent.com/736109/106192989-374d9580-61ad-11eb-97cf-22f1db43e1d9.mov

## Types of changes


## Checklist:



## QA

Checks: https://github.com/owncloud/ios-app/pull/881#issuecomment-795038786

Bugs & Improvements:

- [X] (1) Passcode view always dark https://github.com/owncloud/ios-app/pull/881#issuecomment-795040139
- [X] (2) Brute force protection https://github.com/owncloud/ios-app/pull/881#issuecomment-795049508
- [x] (3) App is displayed after typing correct passcode in Files app (iPad) https://github.com/owncloud/ios-app/pull/881#issuecomment-797480195
- [X] (4) Touch id does not unblock files app https://github.com/owncloud/ios-app/pull/881#issuecomment-799300538
